src/Features/CSharp/Portable/GoToDefinition/CSharpGoToDefinitionSymbolService.cs

```diff
@@ -20,5 +22,77 @@ public CSharpGoToDefinitionSymbolService()
 
         protected override ISymbol FindRelatedExplicitlyDeclaredSymbol(ISymbol symbol, Compilation compilation)
             => symbol;
+
+        protected override int? GetTargetPositionIfControlFlow(SyntaxToken token)
+        {
+            var node = token.GetRequiredParent();
+
+            switch (token.Kind())
+            {
+                case SyntaxKind.ContinueKeyword:
+
+                    var foundContinuedLoop = TryFindContainingLoop(node);
+
+                    return foundContinuedLoop?.IsContinuableConstruct() == true
+                        ? foundContinuedLoop.GetFirstToken().Span.Start
+                        : null;
+
+                case SyntaxKind.BreakKeyword:
+                    if (token.GetPreviousToken().IsKind(SyntaxKind.YieldKeyword))
+                    {
+                        goto case SyntaxKind.YieldKeyword;
+                    }
+
+                    var foundBrokenLoop = TryFindContainingLoop(node);
+
+                    return foundBrokenLoop?.IsBreakableConstruct() == true
+                        ? foundBrokenLoop.GetLastToken().Span.End
+                        : null;
+
+                case SyntaxKind.YieldKeyword:
+                case SyntaxKind.ReturnKeyword:
+                    {
+                        var foundReturnableConstruct = TryFindContainingReturnableConstruct(node);
+                        return foundReturnableConstruct?.GetFirstToken().Span.Start;
+                    }
+            }
+
+            return null;
+
+            static SyntaxNode? TryFindContainingLoop(SyntaxNode? node)
+            {
+                while (node is not null && !node.IsContinuableConstruct() && !node.IsBreakableConstruct())
```

Comment:
this seems wwrong.  say you have a switch inside a loop.  we'll hit the 'switch' as it is breakable.

Also, i swear this code must already exist :)  we have to do the same for keyword highlighting so match things like switch/break and for/continue.

Reply:
Thanks. That was indeed a bug.

There's only a couple other callers for `IsContinuableConstruct`, such as `HighlightRelatedKeywords` and `ContinueKeywordRecommender.IsValidContext`, but they don't seem re-usable for this.